frontend/src/views/main/project/ProjectsHome.vue

```diff
@@ -92,90 +92,100 @@
   </div>
 </template>
 
-<script lang="ts">
-import {Component, Vue, Watch} from 'vue-property-decorator';
-import {ValidationObserver} from 'vee-validate'
-import {readAllProjects, readHasAdminAccess, readUserProfile} from '@/store/main/getters';
-import {Role} from '@/enums';
-import {dispatchCreateProject, dispatchGetProjects} from '@/store/main/actions';
-import {ProjectPostDTO} from '@/generated-sources';
+<script setup lang="ts">
+import {computed, onMounted, ref, watch} from "vue";
+import {ValidationObserver} from "vee-validate";
+import {dispatchCreateProject, dispatchGetProjects} from "@/store/main/actions";
+import {readAllProjects, readHasAdminAccess, readUserProfile} from "@/store/main/getters";
+import {Role} from "@/enums";
+import {ProjectPostDTO} from "@/generated-sources";
+import {toSlug} from "@/utils";
+import store from "@/store";
+import {useRoute, useRouter} from "vue-router/composables";
 import moment from "moment";
-import {toSlug} from '@/utils';
 
-@Component
-export default class ProjectsHome extends Vue {
+const route = useRoute();
+const router = useRouter();
 
-  private openCreateDialog = false; // toggle for edit or create dialog
-  private newProjectName = "";
-  private newProjectKey = "";
-  private newProjectDesc = "";
-  private creatorFilter = 0;
-  private projectCreateError = "";
+const openCreateDialog = ref<boolean>(false); // toggle for edit or create dialog
+const newProjectName = ref<string>("");
+const newProjectKey = ref<string>("");
+const newProjectDesc = ref<string>("");
+const creatorFilter = ref<number>(0);
+const projectCreateError = ref<string>("");
 
-  $refs!: {
-    dialogForm: InstanceType<typeof ValidationObserver>;
-  };
 
-  public async mounted() {
-    const f = this.$route.query.f ? this.$route.query.f[0] || "" : ""
-    this.creatorFilter = parseInt(f) || 0;
-    await this.loadProjects();
-  }
+// template ref
+const dialogForm = ref<InstanceType<typeof ValidationObserver> | null>(null);
 
-  private async loadProjects() {
-    await dispatchGetProjects(this.$store);
-  }
+onMounted(async () => {
+  const f = route.query.f ? route.query.f[0] || '' : '';
+  creatorFilter.value = parseInt(f) || 0;
+  await loadProjects();
+})
 
-  get userProfile() {
-    return readUserProfile(this.$store);
+// computed
+const userProfile = computed(() => {
+  const userProfile = readUserProfile(store);
+  if (userProfile == null) {
+    throw Error("User is not defined.")
   }
+  return userProfile;
+});
 
-  public get isAdmin() {
-    return readHasAdminAccess(this.$store);
-  }
+const isAdmin = computed(() => {
+  return readHasAdminAccess(store);
+});
 
-  public get isCreator() {
-    return this.userProfile?.roles?.includes(Role.AICREATOR);
-  }
+const isCreator = computed(() => {
+  return this.userProfile?.roles?.includes(Role.AICREATOR);
```

Comment:
there's no `this` anymore, plus a computable is a `ref` so to refer to its value you'd need to use
`userProfile.value`:

```suggestion
  return userProfile.value.roles?.includes(Role.AICREATOR);
```